kafka-plugins-common/src/main/java/io/cdap/plugin/batch/source/KafkaBatchConfig.java

```diff
@@ -162,20 +167,21 @@ public String getFormat() {
   }
 
   @Nullable
-  public Schema getSchema() {
+  public Schema getSchema(FailureCollector collector) {
     try {
       return Strings.isNullOrEmpty(schema) ? null : Schema.parseJson(schema);
     } catch (IOException e) {
-      throw new IllegalArgumentException("Unable to parse schema: " + e.getMessage());
+      collector.addFailure("Unable to parse schema: " + e.getMessage(), null).withConfigProperty("schema");
+      return null;
     }
   }
 
   /**
    * Gets the message schema from the schema field.
    * If the time, key, partition, or offset fields are in the configured schema, they will be removed.
    */
-  public Schema getMessageSchema() {
-    Schema schema = getSchema();
+  public Schema getMessageSchema(FailureCollector collector) {
+    Schema schema = getSchema(collector);
```

Comment:
if this has a problem, schema will be null and you'll get a NullPointerException below. Need to do a null check to avoid that.